docs/design.md

```diff
@@ -0,0 +1,49 @@
+# Design notes
+
+## Motivation
+
+To run dctest in kubernetes cluster
+
+### Goals
+
+- Develop the custom controller to run dctest in a pod
+- Automate start and stop dctest
+
+## Architecture & Components
+
+### Kubernetes Custom Resources
+
+#### `VirtualDC`
+
+This is custom resource to deploy dctest pod.
+Nyamber create dctest pod and service from its definition.
+
+#### `AutoVirtualDC`
+
+This is custom resource to automate deployment dctest pod.
+Nyamber create `VirtualDC` custom resource from its definition according to the schedule field.
+You can specify the `VirtualDC` startSchedule and stopSchedule with cron format.
+
+
+### Kubernetes workloads
+
+#### `virtualdc-controller`
+
+A deployment that controls Runner Pod.
+It create Runner pod which runs dctest and service to access Runner pod.
+
+#### `autovirtualdc-controller`
+
+A deployment that create and delete VirutualDC resources according to set schedules.
```

Comment:
In addition to schedule, it would be nice to mention retries in case of dctest bootstrap failure.